frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -244,6 +244,20 @@ scopeResolveModule(Context* context, ID id) {
 
 }
 
+// Find the module-level statement ID
+static ID findModuleStmtId(Context* context, ID id) {
+  ID curId = id;
+  ID parentId = parsing::idToParentId(context, curId);
+  const AstNode* parent = parsing::idToAst(context, parentId);
```

Comment:
The way this is written will result in causing `typeForModuleLevelSymbol` not to be reused, but to be recomputed, if anything at all inside the entire module changes. I don't think that's what we want.

How about using `idToContainingMultiDeclId` instead, which AFAIK does what you want more directly and also avoids this problem?